src/bluesearch/entrypoint/database/run.py

```diff
@@ -0,0 +1,548 @@
+    """Add parsed articles to the database.
+
+    This step is considered done if all articles inside of `parsed/` are
+    already in the database.
+    """
+
+    db_url = luigi.Parameter()
+    db_type = luigi.Parameter()
+
+    @staticmethod
+    def requires() -> luigi.Task:
+        """Define dependency."""
+        return ParseTask()
+
+    def complete(self) -> bool:
+        """Check if all articles inside of `parsed/` are in the database."""
+        # If all the articles are inside
+        if self.db_type == "sqlite":
+            prefix = "sqlite:///"
+        elif self.db_type == "postgres":
+            prefix = "postgresql+pg8000://"
+        else:
+            raise ValueError
+
+        engine = sqlalchemy.create_engine(f"{prefix}{self.db_url}")
+
+        input_dir = Path(self.input().path)
+        if not input_dir.exists():
+            return False
+
+        all_uids = [
+            article.stem for article in input_dir.iterdir() if article.suffix == ".json"
+        ]
+
+        new_uids = []
+        for uid in all_uids:
+            query = sqlalchemy.text(
+                "SELECT article_id from articles WHERE article_id = :uid"
+            )
+            res = engine.execute(query, uid=uid).fetchall()
+
+            if not res:
+                new_uids.append(uid)
+
+        return not new_uids
+
+    def program_args(self) -> list[str]:
+        """Define subprocess arguments."""
+        input_dir = Path(self.input().path)
+
+        command = [
+            *BBS_BINARY,
+            "add",
+            *VERBOSITY,
+            self.db_url,
+            input_dir,
+            f"--db-type={self.db_type}",
+        ]
+
+        return command
+
+
+def run(
+    *,
+    dry_run: bool,
+    final_task: str | None = None,
+    config_path: Path | None = None,
+    luigi_config: str | None = None,
+) -> int:
+    """Run overall pipeline.
+
+    Parameter description and potential defaults are documented inside of the
+    `get_parser` function.
+    """
+    logger.info("Starting the overall pipeline")
+
+    DownloadTask.capture_output = CAPTURE_OUTPUT
+    UnzipTask.capture_output = CAPTURE_OUTPUT
+    TopicExtractTask.capture_output = CAPTURE_OUTPUT
+    TopicFilterTask.capture_output = CAPTURE_OUTPUT
+    PerformFilteringTask.capture_output = CAPTURE_OUTPUT
+    ConvertPDFTask.capture_output = CAPTURE_OUTPUT
+    ParseTask.capture_output = CAPTURE_OUTPUT
+    AddTask.capture_output = CAPTURE_OUTPUT
+
+    if config_path:
+        if not pathlib.Path(config_path).exists():
+            raise ValueError(f"The configuration path {config_path} does not exist!")
+
+        config = luigi.configuration.get_config()
+        config.add_config_path(config_path)
+        config.reload()
+
+    if luigi_config:
+        config = luigi.configuration.get_config()
+        for param in luigi_config.split(","):
+            change = re.split(r"[.:]", param, maxsplit=3)
+            config.set(*change)
```

Comment:
Do we need both `config-path` and `luigi-config` CLI arguments? I am not sure I understand the difference.

Also, isn't it easier/better to remove them and just rely on the env variable `LUIGI_CONFIG_PATH` (see [`luigi` Docs](https://luigi.readthedocs.io/en/stable/configuration.html#configuration))? This is for instance what BBP's `mcar` does, see e.g. how we are invoking it [here ](https://bbpgitlab.epfl.ch/ml/deepmorphs/-/blob/main/dvc/dvc.lock#L9149) in another project.
Then we would be able to run the entrypoint like this:
```bash
LUIGI_CONFIG_PATH=some/path/luigi.cfg bbs_database run [args...]
```

Reply:
- `config-path` is there to change the path to the configuration file. 
- `luigi-config` is to allow the user to overwrite a parameter (if needed). e.g. `--luigi-config GlobalParams.source:arxiv`.

If you prefer, I can remove those two arguments. Maybe just a side note related to this: I have the feeling it is much easier to tests the complete pipeline with those two arguments.

Reply:
Suggestions:
- [ ] Check what happens if a user provides `config-path` but then `LUIGI_CONFIG_PATH` is also set as an environmental variable. What has precedence?
- [ ] Rename `config-path` as `luigi-config-path` + make parser helper more explicit (`help="Configuration Path."` --> `help="Path to Luigi configuration file."`)
- [ ] Rename `luigi-config` as `luigi-config-args`  + make parser helper more explicit (`help="Configuration parameters."` --> `help="Comma separated key-value arguments for Luigi configuration, e.g. "--luigi-config GlobalParams.source:arxiv,DownloadTask.from-month:2021-04" . Overwrites the content of Luigi configuration file (see --luigi-config-path)."`)

Reply:
Done in [0d343d3](https://github.com/BlueBrain/Search/pull/571/commits/0d343d3af274de31c082bb00259dd1f9a70864f1).
For info, the `luigi-config-path`has precedence over the environment variable.